src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/TextFieldParser.vb

```diff
@@ -0,0 +1,1602 @@
+            m_Reader = New StreamReader(fileStreamTemp, defaultEncoding, detectEncoding)
+
+            ReadToBuffer()
+        End Sub
+
+        ''' <summary>
+        '''  Creates a StreamReader for a passed in stream
+        ''' </summary>
+        ''' <param name="stream">The passed in stream</param>
+        ''' <param name="defaultEncoding">The encoding to default to if encoding can't be detected</param>
+        ''' <param name="detectEncoding">Indicates whether or not to detect encoding from the BOM</param>
+        ''' <remarks>We validate the arguments here for the three Public constructors that take a Stream</remarks>
+        Private Sub InitializeFromStream(ByVal stream As Stream, ByVal defaultEncoding As System.Text.Encoding, ByVal detectEncoding As Boolean)
+            If stream Is Nothing Then
+                Throw GetArgumentNullException("stream")
+            End If
+
+            If Not stream.CanRead Then
+                Throw GetArgumentExceptionWithArgName("stream", SR.TextFieldParser_StreamNotReadable, "stream")
+            End If
+
+            If defaultEncoding Is Nothing Then
+                Throw GetArgumentNullException("defaultEncoding")
+            End If
+
+            m_Reader = New StreamReader(stream, defaultEncoding, detectEncoding)
+            ReadToBuffer()
+        End Sub
+
+        ''' <summary>
+        '''  Gets full name and path from passed in path.
+        ''' </summary>
+        ''' <param name="path">The path to be validated</param>
+        ''' <returns>The full name and path</returns>
+        ''' <remarks>Throws if the file doesn't exist or if the path is malformed</remarks>
+        Private Shared Function ValidatePath(ByVal path As String) As String
+            ' Validate and get full path
+            Dim fullPath As String = FileSystem.NormalizeFilePath(path, "path")
+
+            ' Make sure the file exists
+            If Not File.Exists(fullPath) Then
+                Throw New IO.FileNotFoundException(GetResourceString(SR.IO_FileNotFound_Path, fullPath))
+            End If
+            Return fullPath
+        End Function
+
+        ''' <summary>
+        '''  Indicates whether or not the passed in line should be ignored
+        ''' </summary>
+        ''' <param name="line">The line to be tested</param>
+        ''' <returns>True if the line should be ignored, otherwise False</returns>
+        ''' <remarks>Lines to ignore are blank lines and comments</remarks>
+        Private Function IgnoreLine(ByVal line As String) As Boolean
+            ' If the Line is Nothing, it has meaning (we've reached the end of the file) so don't
+            ' ignore it
+            If line Is Nothing Then
+                Return False
+            End If
+
+            ' Ignore empty or whitespace lines
+            Dim TrimmedLine As String = line.Trim()
+            If TrimmedLine = "" Then
```

Comment:
>TrimmedLine = "" [](start = 15, length = 16)

`TrimmedLine.Length = 0` to match reference source. (There's no need to check for `Nothing` in this case.)